src/WindowsTracing/Tracer.cpp

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "WindowsTracing/Tracer.h"
+
+#include "KrabsTracer.h"
+#include "OrbitBase/Logging.h"
+
+namespace orbit_windows_tracing {
+
+Tracer::Tracer(orbit_grpc_protos::CaptureOptions capture_options, TracerListener* listener)
+    : capture_options_(std::move(capture_options)), listener_(listener) {}
+
+void Tracer::Start() {
+  CHECK(tracer_impl_ == nullptr);
+  tracer_impl_ = std::make_unique<KrabsTracer>(capture_options_, listener_);
```

Comment:
I am a bit confused about this implementation. Shouldn't Tracer be simply an interface, and we have KrabsTracer as subclass, implementing Start/Stop, and maybe provide a Factory for the concrete tracer?

Reply:
Yeah, `tracer_impl_` is a field holding an instance of a subclass, something seems wrong. Either `KrabsTracer` is a subclass and `Tracer` a pure interface, or `KrabsTracer` is a "normal" class that `Tracer` forwards to.

Reply:
I want to make sure that even if a user holds on to a Tracer object for multiple Start/Stop cycles, a new KrabsTracer object will be created for each session, avoiding the need for resetting the KrabsTracer explicitly, and clearing the memory on session Stop. But yeah, I agree, the current setup is weird, will rework.

Reply:
Introduced a TracerInterface in WindowsTracing. Both `Tracer` and `KrabsTracer` implement the interface. Tracer also owns an opaque pointer to a `TracerInterface`, which internally is set to a KrabsTracer instance on "Start". This is done to hide the implementation and also make sure that a new KrabsTracer object is created on every profiling session, and that we clear its memory on "Stop".